src/libraries/Common/src/Interop/Unix/System.Native/Interop.ReadLink.cs

```diff
@@ -21,7 +21,7 @@ internal static partial class Sys
         /// Returns the number of bytes placed into the buffer on success; bufferSize if the buffer is too small; and -1 on error.
         /// </returns>
         [LibraryImport(Libraries.SystemNative, EntryPoint = "SystemNative_ReadLink", SetLastError = true)]
```

Comment:
This has `SetLastError`, but there are multiple places where the last error can be overwritten before it is read.

For example, `ArrayPool<byte>.Shared.Return` can overwrite the last error.

Reply:
It is used here :
https://github.com/dotnet/runtime/blob/8e8a62156ea459945b1d953c4fbe950d276bbc9b/src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs#L629-L640

And this breaks many unit tests where we throw exceptions with paths that are not symbolic link

Reply:
@jkotas Does @pedrobsaila's comment address your concern on this?

https://github.com/dotnet/runtime/pull/69335/files#r875233856

Reply:
Yes.